Source/WebCore/xml/XPathParser.cpp

```diff
@@ -47,16 +47,31 @@ namespace XPath {
 
 struct Parser::Token {
     int type;
-    String string;
-    Step::Axis axis;
-    NumericOp::Opcode numericOpcode;
-    EqTestOp::Opcode equalityTestOpcode;
-
-    Token(int type) : type(type) { }
-    Token(int type, const String& string) : type(type), string(string) { }
-    Token(int type, Step::Axis axis) : type(type), axis(axis) { }
-    Token(int type, NumericOp::Opcode opcode) : type(type), numericOpcode(opcode) { }
-    Token(int type, EqTestOp::Opcode opcode) : type(type), equalityTestOpcode(opcode) { }
+    using TokenValue = std::variant<String, Step::Axis, NumericOp::Opcode, EqTestOp::Opcode>;
+    TokenValue value;
+
+    Token() = delete;
+
+    Token(int type)
+        : type(type)
+    { }
+    Token(int type, const String& string)
```

Comment:
Maybe we can take a `const TokenValue&` as parameter and require only a single constructor instead of 4?

Reply:
Great idea!  I replaced all four constructors with this:

```
    Token(int type, TokenValue&& value)
        : type(type), value(WTFMove(value))
    { }
```